chain/contractWrappers.go

```diff
@@ -233,9 +233,36 @@ func (ph *VotingHandle) SetStatusTxArgs(ctx context.Context, pid [types.ProcessI
 	return setprocessTxArgs, nil
 }
 
-// SetCensusTxArgs
-func (ph *VotingHandle) SetCensusTxArgs(ctx context.Context, pid [types.ProcessIDsize]byte) (*models.SetProcessTx, error) {
-	return nil, fmt.Errorf("not implemented")
+// SetCensusTxArgs returns a SetProcess tx instance with census mkroot and census mkURI set
+func (ph *VotingHandle) SetCensusTxArgs(ctx context.Context, pid [types.ProcessIDsize]byte, namespace uint16) (*models.SetProcessTx, error) {
+	processData, err := ph.VotingProcess.Get(&ethbind.CallOpts{Context: ctx}, pid)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching process from Ethereum: %w", err)
+	}
+	// create setProcessTx
+	setprocessTxArgs := new(models.SetProcessTx)
+	// process id
+	setprocessTxArgs.ProcessId = pid[:]
+	// process mkRoot
+	if setprocessTxArgs.CensusRoot, err = hex.DecodeString(util.TrimHex(processData.MetadataCensusRootCensusUri[1])); err != nil {
+		return nil, fmt.Errorf("invalid mkRoot: %w", err)
+	}
+	// process mkUri
+	// check origin
+	switch t := models.CensusOrigin(processData.ModeEnvelopeTypeCensusOrigin[2]); {
+	case t == models.CensusOrigin_OFF_CHAIN_CA, t == models.CensusOrigin_OFF_CHAIN_TREE, t == models.CensusOrigin_OFF_CHAIN_TREE_WEIGHTED:
+		if types.CensusOrigins[t].NeedsURI && len(processData.MetadataCensusRootCensusUri[2]) == 0 {
+			return nil, fmt.Errorf("census %s needs URI but non has been provided", t.String())
+		}
+		setprocessTxArgs.CensusURI = &processData.MetadataCensusRootCensusUri[2]
+	default:
+		return nil, fmt.Errorf("invalid census origin, only off-chain processes can update the census")
```

Comment:
I'd remove `only off-chain processes can update the census` since its true now but in might not be in the future.

Reply:
yep, it's more flexible, changed!